### DISCUSSION TITLE
Descope Header

### DISCUSSION THREAD

**Discussion by BlackF0rest** [zauberzeug/nicegui#2348] [Q&A]
### Question

Hey, 
i am currently writing a website-scraper wich displays its contents in a nicegui gui. Im using a descope authentication which works really well. The only problem i have is that when using the descope example with the page class u cannot add a header anymore. Is there any way i can still do that even tho the client is allready awaited the clients connection? 
Here is the Comment from the descope user.py example code:
```py
"""A page that requires the user to be logged in.

It allows the same parameters as ui.page, but adds a login check.
As recommended by Descope, this is done via JavaScript and allows to use Flows.
But this means that the page has already awaited the client connection.
So `ui.add_head_html` will not work.
"""
```

**Comment by falkoschindler** [comment]
Hi @BlackF0rest,

this docstring is kind of outdated, because [NiceGUI 1.4.9](https://github.com/zauberzeug/nicegui/releases/tag/v1.4.9) added support for calling `ui.add_head_html` and `ui.run_javascript` after the client connected. This means we should be able to remove the line `await client.connected()`. Can you try that out? If it works we can update the example.

**Reply by BlackF0rest** [reply]
Hi, 
thanks for the fast reply. 
When i just comment out the `await client.connected()` in the example it throws the following error:
`Cannot open page because client is not connected, try RedirectResponse from FastAPI instead`

When comment out the line in my own code with a `ui.header()` in the code it throws the following error:
`RuntimeError: Cannot run JavaScript before client is connected; try "await client.connected()" or "client.on_connect(...)".`

**Reply by falkoschindler** [reply]
Ok, the message "Cannot run JavaScript before client is connected" indicates that you don't use version 1.4.9.
But `ui.open` still seems to require a socket connection. We should fix that.

**Reply by falkoschindler** [reply]
We just released version [1.4.11](https://github.com/zauberzeug/nicegui/releases/tag/v1.4.11) with this fix.

**Reply by BlackF0rest** [reply]
Hi, updated to the newest version and tried it out on the example but there is this error now:
`JavaScript did not respond within 1.0 s`
when just commenting out the `#await client.conencted()`

**Reply by BlackF0rest** [reply]
Okay but when i run my code with the await it still works fine. So original problem solved!! Thank you.
